src/java.base/share/classes/java/lang/foreign/MemorySegment.java

```diff
@@ -1312,6 +1312,44 @@ MemorySegment reinterpret(long newSize,
      */
     String getString(long offset, Charset charset);
 
+    /**
+     * Reads a string using the given byte length from this segment at the given offset,
```

Comment:
What happens is there's a null terminator before `length` ?

Reply:
Overall, I guess I'm not overly convinced that it's worth adding a method like this for:
```
byte[] bytes = new byte[len];
MemorySegment.copy(segment, JAVA_BYTE, offset, bytes, 0, len);
return new String(bytes, charset);
```
e.g. the API provides string shortcuts for the common case. The more convoluted case can still be achieve via `copy`. Is there anything I'm missing here?

Reply:
Ok, I suppose what you do get is the auto-detection of charset length, which adjusts the size of the copy...

Reply:
> Ok, I suppose what you do get is the auto-detection of charset length, which adjusts the size of the copy...

NVM, this is not what this impl is doing. I see no performance benefit for using the new API over a `copy`. If so, this new API is just for ease of use. While that's ok, IMHO it's not an easy API to add, as we're making a number of arbitrary choices here:
* what to do with terminators in the middle of the string
* whether length is length of string in chars, or length in bytes (we do the latter)

This leaves me a bit skeptical that this API is the right primitive.